jetty-core/jetty-jmx/src/main/java/org/eclipse/jetty/jmx/ConnectorServer.java

```diff
@@ -182,7 +184,20 @@ public void doStart() throws Exception
         {
             int startIndex = jndiRMI.length();
             int endIndex = urlPath.indexOf('/', startIndex);
-            HostPort hostPort = new HostPort(urlPath.substring(startIndex, endIndex));
+            String rawHost = urlPath.substring(startIndex, endIndex);
+            HostPort hostPort;
+
+            if (StringUtil.isBlank(rawHost)) // no host
+            {
+                hostPort = new HostPort(InetAddress.getLocalHost().getHostName(), DEFAULT_PORT);
+            }
+            else if (rawHost.startsWith(":")) // port without host
+            {
+                hostPort = new HostPort(InetAddress.getLocalHost().getHostName() + rawHost);
+            }
+            else
+                hostPort = new HostPort(rawHost);
```

Comment:
Drop the braces, since they are all one-liners.

Also, `getHostName()` performs a reverse DNS lookup -- use `getHostAddress()` instead?

Reply:
Done.